perceval/components/processor.py

```diff
@@ -282,7 +285,7 @@ def probs(self, precision: float = None, progress_callback: callable = None) ->
             self._simulator.set_precision(precision)
         get_logger().info(f"Start a local {'perfect' if self._source.is_perfect() else 'noisy'} strong simulation",
                           channel.general)
-        res = self._simulator.probs_svd(self._inputs_map, progress_callback, self._detectors)
+        res = self._simulator.probs_svd(self._inputs_map, self._detectors, progress_callback)
```

Comment:
Why does the sampling function use `set_detectors` and not the prob function?
and this would remove the code breaking

Reply:
We discussed yesterday that it made sense to pass the detectors only to `probs_svd` in the strong simulation class, as detectors would be ignored in the majority of the calls (and currently only used in this one - probs_svd). So, I'm still for this way of injecting detectors in the different simulators.